include/seqan3/search/configuration/hit.hpp

```diff
@@ -90,4 +93,71 @@ struct hit_strata : public pipeable_config_element<hit_strata, uint8_t>
     static constexpr detail::search_config_id id{detail::search_config_id::hit};
 };
 
+/*!\brief A dynamic configuration element to configure the hit strategy at runtime.
+ * \ingroup search_configuration
+ * \sa \ref search_configuration_subsection_hit_strategy "Section on Hit Strategy"
+ */
+class hit : public pipeable_config_element<hit, seqan3::detail::empty_type>
+{
+public:
+    //!\brief The type of the std::variant holding the hit configuration element alternatives.
+    using hit_variant_type = std::variant<detail::empty_type,
```

Comment:
```suggestion
    /*!\brief The type of the std::variant holding the hit configuration element alternatives
     *
     * The additional detail::empty_type marks the seqan3::search_cfg::hit as default constructed, with no selected hit
     * configuration which can be checked within the search algorithm.
     */
    using hit_variant_type = std::variant<detail::empty_type,
```
Or something along these lines. because it is public and the user might wonder what empty_type stands for.

Reply:
I already have a ticket to make empty type public since we are using it now quite a lot in a public API.